### PR TITLE
bugfix: the disposables tree set won't accept another Disposable with the same priority.

### PR DESCRIPTION
**Pull request by jujinghao**
Fix the following issue:
Disposable TmNettyRemotingClient and RmNettyRemotingClient added by GlobalTransactionScanner have the same priority by default, so the RmNettyRemotingClient won't be closed gracefully during shutdown.

### CODE REVIEW

**Review by slievrly — APPROVED** [review]
LGTM, great found.

File: core/src/main/java/io/seata/core/rpc/ShutdownHook.java [hunk 1]
```diff
@@ -91,35 +91,63 @@ public static void removeRuntimeShutdownHook() {
...
+
         public DisposablePriorityWrapper(Disposable disposable, int priority) {
             this.disposable = disposable;
             this.priority = priority;
+            this.seqId = seq.incrementAndGet();
         }
 
         @Override
         public int compareTo(DisposablePriorityWrapper disposablePriorityWrapper) {
-            return priority - disposablePriorityWrapper.priority;
```
Comment (selfishlover) [inline core/src/main/java/io/seata/core/rpc/ShutdownHook.java:105]: use PriorityQueue to replace TreeSet will solve this problem once and forever. No need to add a "seqId" to fix the probem which is actually caused by TreeSet.

File: core/src/main/java/io/seata/core/rpc/ShutdownHook.java [hunk 2]
```diff
@@ -91,35 +91,63 @@ public static void removeRuntimeShutdownHook() {
...
+            int cmp = priority - disposablePriorityWrapper.priority;
+            if (cmp == 0) {
+                if (seqId > disposablePriorityWrapper.seqId) {
+                    cmp = 1;
+                } else if (seqId < disposablePriorityWrapper.seqId) {
+                    cmp = -1;
+                } else {
+                    cmp = 0;
+                }
+            }
```
Comment (jsbxyyx) [inline core/src/main/java/io/seata/core/rpc/ShutdownHook.java:119]: Long.compare(x, y) ?

File: core/src/main/java/io/seata/core/rpc/ShutdownHook.java [hunk 3]
```diff
@@ -91,35 +91,63 @@ public static void removeRuntimeShutdownHook() {
...
         }
 
         @Override
         public int hashCode() {
-            return Objects.hashCode(this.priority);
+            final int prime = 31;
+            int result = 1;
+            result = prime * result + priority;
+            result = prime * result + (int) (seqId ^ (seqId >>> 32));
+            return result;
```
Comment (jsbxyyx) [inline core/src/main/java/io/seata/core/rpc/ShutdownHook.java:129]: Objects.hash(...values)?

**Review by l81893521 — APPROVED** [review]
LGTM

File: core/src/main/java/io/seata/core/rpc/ShutdownHook.java (hunk 3 above)
Comment (a364176773) [reply core/src/main/java/io/seata/core/rpc/ShutdownHook.java:129]: +1

File: core/src/main/java/io/seata/core/rpc/ShutdownHook.java (hunk 2 above)
Comment (a364176773) [reply core/src/main/java/io/seata/core/rpc/ShutdownHook.java:119]: +1

**Review by a364176773 — APPROVED** [review]
LGTM

**Pull request merged**